v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/AvroToValueMapperTest.java

```diff
@@ -168,6 +168,68 @@ public void testAvroFieldToDouble_UnsupportedType() {
     AvroToValueMapper.avroFieldToDouble(inputValue, SchemaBuilder.builder().booleanType());
   }
 
+  @Test
+  public void testAvroFieldToFloat_ValidFloat32Input() {
+    Float inputValue = 5.75f;
+    Float result =
+        AvroToValueMapper.avroFieldToFloat32(inputValue, SchemaBuilder.builder().doubleType());
+    assertEquals(inputValue, result);
+
+    /* Test a valid double to float mapping */
+    result =
+        AvroToValueMapper.avroFieldToFloat32((Double) 3.14, SchemaBuilder.builder().floatType());
+    assertEquals("Test float input", (Float) 3.14f, result);
+
+    result = AvroToValueMapper.avroFieldToFloat32("456.346", SchemaBuilder.builder().doubleType());
+    assertEquals("Test string input", (Float) 456.346f, result);
+
+    result =
+        AvroToValueMapper.avroFieldToFloat32(Float.MAX_VALUE, SchemaBuilder.builder().doubleType());
+    assertEquals("Test max Float", (Float) Float.MAX_VALUE, result);
+
+    result =
+        AvroToValueMapper.avroFieldToFloat32(Float.MIN_VALUE, SchemaBuilder.builder().doubleType());
+    assertEquals("Test min Float", (Float) Float.MIN_VALUE, result);
+
+    result = AvroToValueMapper.avroFieldToFloat32(Float.NaN, SchemaBuilder.builder().doubleType());
+    assertEquals("Test nan Float", (Float) Float.NaN, result);
+
+    result =
+        AvroToValueMapper.avroFieldToFloat32(
+            Float.POSITIVE_INFINITY, SchemaBuilder.builder().doubleType());
+    assertEquals("Test positive infinity Float", (Float) Float.POSITIVE_INFINITY, result);
+
+    result =
+        AvroToValueMapper.avroFieldToFloat32(
+            Float.NEGATIVE_INFINITY, SchemaBuilder.builder().doubleType());
+    assertEquals("Test negetive infinity Float", (Float) Float.NEGATIVE_INFINITY, result);
+
+    result =
+        AvroToValueMapper.avroFieldToFloat32(Long.MAX_VALUE, SchemaBuilder.builder().longType());
```

Comment:
long can store 64 bits. I'm not sure it fits in float 32

Reply:
Thanks. `Long.MAX_VALUE` came in as I used it as a template to generate float.max and min by search replace.
Removed the same.